Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -200,9 +203,21 @@ function Publish-UpgradedServiceFabricApplication
 
         $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }
         if ($reg)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            $reg | Unregister-ServiceFabricApplicationType -Force
+        {            
+            try {
+                Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))            
+                $reg | Unregister-ServiceFabricApplicationType -Force    
+            }
+            catch [System.Fabric.FabricException] {
+                ## If SkipUpgrade param is set when existing version already deployed, then back out of upgrade
+                if($SkipUpgradeSameTypeAndVersion -And $Error[0].Exception.Message -like '*Application type and version is still in use*') {                    
```

Comment:
Checking based on the error message won't work on a non-english machine and will break if the Service Fabric platform changes their error messages. Have you looked at the [Error Code property](https://docs.microsoft.com/en-us/dotnet/api/system.fabric.fabricexception#System_Fabric_FabricException_ErrorCode)? It looks like the value ApplicationAlreadyInTargetVersion may be what you are looking for.